packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -363,6 +408,55 @@ class LengthLimitingTextInputFormatter extends TextInputFormatter {
   /// composing is not allowed.
   final int? maxLength;
 
+  /// Determines how the [maxLength] limit should be enforced.
+  ///
+  /// Defaults to [MaxLengthEnforcement.enforced].
+  ///
+  /// {@macro flutter.services.textFormatter.maxLengthEnforcement}
+  final MaxLengthEnforcement? maxLengthEnforcement;
+
+  /// Return an effective [MaxLengthEnforcement] according the target platform.
+  ///
+  /// If [platform] was not specified, [defaultTargetPlatform] will be used.
+  ///
+  /// {@template flutter.services.textFormatter.effectiveMaxLengthEnforcement}
+  /// ### Platform specific behaviors
+  ///
+  /// Different platforms follow different behaviors by default, according to
+  /// their native behavior.
+  ///  * Android, Windows: [MaxLengthEnforcement.enforced]. The native behavior
+  ///    of these platforms is enforced. The composing will be handled by the
+  ///    IME while users are entering CJK characters.
+  ///  * iOS: [MaxLengthEnforcement.truncateAfterCompositionEnds]. iOS has no
+  ///    default behavior and it requires users implement the behavior
+  ///    themselves. Allow the composition to exceed to avoid breaking CJK input.
+  ///  * Web, macOS, linux, fuchsia:
+  ///    [MaxLengthEnforcement.truncateAfterCompositionEnds]. These platforms
+  ///    allow the composition to exceed by default.
+  /// {@endtemplate}
+  static MaxLengthEnforcement defaultMaxLengthEnforcement(
+    MaxLengthEnforcement? previousEnforcement, {
+    TargetPlatform? platform,
+  }) {
+    if (previousEnforcement != null) {
+      return previousEnforcement;
+    }
+    if (kIsWeb) {
+      return MaxLengthEnforcement.truncateAfterCompositionEnds;
```

Comment:
Yes but the composing range will be displayed directly on the input element.

Reply:
It's generally up to the IME to decide what needs to be placed in the input element. I tried this on a web page using chrome, on an Android phone, it has the same behavior as in a native app.

Reply:
Okay so I think we can make it (web) enforced as default. What do you think? FYI @justinmc

Reply:
Oh wait, I remember that web has events about compositions, like "compositionStart" and "compositionEnd"? If that's true I think it should still be allow composing since it actually replies on how developer handles event in native web.

Reply:
Here is an example we've discussed before, that not only the IME might handles itself, but the input element too. https://github.com/flutter/flutter/issues/67898#issuecomment-726452539
And docs https://developer.mozilla.org/en-US/docs/Web/API/InputEvent/isComposing , https://developer.mozilla.org/en-US/docs/Mozilla/IME_handling_guide

Reply:
![Screenshot_20201209-130845](https://user-images.githubusercontent.com/15884415/101587726-c21cda00-3a1f-11eb-94fa-c49fbf6ecd32.jpg)

That's allow composition too.

Reply:
Yeah, both iOS and Android on the web seem to allow composition for me (maxLength is 10):

| iOS Web | Android Web |
| --------- | ------------- |
| <img height="500" src="https://user-images.githubusercontent.com/389558/101672801-2c774f80-3a0b-11eb-8ca6-a783d7c3e22f.PNG" /> | <img height="500" src="https://user-images.githubusercontent.com/389558/101673492-15852d00-3a0c-11eb-9011-307ca717ba45.jpg" /> |

The difference is that when typing Chinese on Android, the composition happens on the keyboard and not in the input.  But when composing English on Android, you can see that the composition is able to overflow.

Reply:
Oh thanks for the clarification.